opentracing-spring-cloud-starter-jaeger/README.md

```diff
@@ -0,0 +1,86 @@
+## Configuration options
+
+All the available configuration options can be seen in `io.opentracing.contrib.spring.cloud.starter.jaeger.JaegerConfigurationProperties`.
+The prefix to be used for these properties is `opentracing.jaeger`.
+
+Beware to use the correct syntax for properties that are camel-case in `JaegerConfigurationProperties`.
+
+* For properties / yaml files use `-`. For example `opentracing.jaeger.log-spans=true`
+* For environment variables use `_`. For example `OPENTRACING_JAEGER_LOG_SPANS` 
+
+## Defaults
+
+If no configuration options are changed and the user does not manually provide any of the beans that the 
+auto-configuration process provides, the following defaults are used:
+
+* `unknown-spring-boot` Will be used as the service-name if no value has been specified to the property `spring.application.name`. 
+* A `CompositeReporter` is provided which does not contain any delegates - effectively functioning as a Noop `Reporter`
+* A `ConstSampler` with the value of `true`. This means that every trace will be sampled
+* A `NoopMetricsFactory` is used - effectively meaning that no metrics will be collected
+
+## Manual bean provisioning
+
+Any of the following beans can be provided by the application (by adding configuring them as bean with `@Bean` for example)
+and will be used to by the Tracer instead of the auto-configured beans.
+
+* `com.uber.jaeger.samplers.Sampler`  
+* `com.uber.jaeger.metrics.MetricsFactory`  
+* `com.uber.jaeger.metrics.Metrics`  
+* `com.uber.jaeger.reporters.Reporter`
+
+## Common cases
```

Comment:
I would split this into component-based settings:

## Reporter  configuration
### Http Sender
* fo.bar.baz. t
### UDP
### Log spans
- can be conbined with others
.. 
## Sampling